components/tag-list/tag-list-item-mixin.js

```diff
@@ -212,12 +218,10 @@ export const TagListItemMixin = superclass => class extends LocalizeCoreElement(
 		`;
 	}
 
-	_renderTag(tagContent, options) {
-		if (!options) options = {};
+	_renderTag(tagContent, options = {}) {
+		this._displayText = options.displayText || (tagContent?.constructor === String ? tagContent : '');
```

Comment:
Maybe we just make `displayText` (or maybe `summaryText` to really differentiate it from the tag's text) a required second parameter? Would be a breaking change but feels a little cleaner, since we do _need_ something to announce and use as the accessible text for the icon button